services/laas/pipeline.go

```diff
@@ -0,0 +1,169 @@
+	apiResponse.LogInfo()
+	return pipelineResponse, apiResponse, err
+}
+
+func (c *Client) DeletePipeline(ctx context.Context, id string) (utils.ApiResponseInfo, error) {
+	_, apiResponse, err := c.sdkClient.PipelinesApi.PipelinesDelete(ctx, id).Execute()
+	apiResponse.LogInfo()
+	return apiResponse, err
+}
+
+func (c *Client) IsPipelineDeleted(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	_, apiResponse, err := c.sdkClient.PipelinesApi.PipelinesFindById(ctx, d.Id()).Execute()
+	apiResponse.LogInfo()
+	return apiResponse.HttpNotFound(), err
+}
+
+func (c *Client) GetPipelineById(ctx context.Context, id string) (laas.Pipeline, *laas.APIResponse, error) {
+	pipeline, apiResponse, err := c.sdkClient.PipelinesApi.PipelinesFindById(ctx, id).Execute()
+	apiResponse.LogInfo()
+	return pipeline, apiResponse, err
+}
+
+func (c *Client) ListPipelines(ctx context.Context) (laas.PipelineListResponse, *laas.APIResponse, error) {
+	pipelines, apiResponse, err := c.sdkClient.PipelinesApi.PipelinesGet(ctx).Execute()
+	apiResponse.LogInfo()
+	return pipelines, apiResponse, err
+}
+
+func setPipelinePostRequest(d *schema.ResourceData) *laas.CreateRequest {
+	request := laas.CreateRequest{Properties: &laas.CreateRequestProperties{}}
+
+	if nameValue, ok := d.GetOk("name"); ok {
+		name := nameValue.(string)
+		request.Properties.Name = &name
+	}
+
+	var logs []laas.CreateRequestPipeline
+	if logsValue, ok := d.GetOk("log"); ok {
+		for _, logData := range logsValue.(*schema.Set).List() {
+			if log, ok := logData.(map[string]interface{}); ok {
+				logSource := log["source"].(string)
+				logTag := log["tag"].(string)
+				logProtocol := log["protocol"].(string)
+				newLog := *laas.NewCreateRequestPipeline()
+				newLog.Source = &logSource
+				newLog.Tag = &logTag
+				newLog.Protocol = &logProtocol
+				logs = append(logs, newLog)
+			}
+		}
+	}
+
+	request.Properties.Logs = &logs
+
+	return &request
+}
+
+func setPipelinePatchRequest(d *schema.ResourceData) *laas.PatchRequest {
+	request := laas.PatchRequest{Properties: &laas.PatchRequestProperties{}}
+
+	if nameValue, ok := d.GetOk("name"); ok {
+		name := nameValue.(string)
+		request.Properties.Name = &name
+	}
+
+	var logs []laas.PatchRequestPipeline
+	if logsValue, ok := d.GetOk("log"); ok {
+		for _, logData := range logsValue.(*schema.Set).List() {
+			if log, ok := logData.(map[string]interface{}); ok {
+				logSource := log["source"].(string)
+				logTag := log["tag"].(string)
+				logProtocol := log["protocol"].(string)
+				newLog := *laas.NewPatchRequestPipeline()
+				newLog.Source = &logSource
+				newLog.Tag = &logTag
+				newLog.Protocol = &logProtocol
+				logs = append(logs, newLog)
+			}
+		}
+	}
+
+	request.Properties.Logs = &logs
+
+	return &request
+}
+
+func (c *Client) SetPipelineData(d *schema.ResourceData, pipeline laas.Pipeline) error {
+	if pipeline.Id != nil {
```

Comment:
we should either exit here if pipeline.id can be nil, or not check for pipeline.Id if it cannot be nul since we deref it below without checking

Reply:
@cristiGuranIonos done in d8c4b47a210fc42a7af535eb2d82ab38246381bd, I removed the check.